### PR TITLE
dedupe IsNodeConditionSetAsExpected test function

### LINKED ISSUE #93965: isNodeConditionSetAsExpected code duplicated, needs comments
**What happened**:

isNodeConditionSetAsExpected is duplicated in two places, and needs a godoc

- test/integration/framework/util.go 
- test/e2e/framework/node/resource.go

**What you expected to happen**:

The function would only be in one place.

**How to reproduce it (as minimally and precisely as possible)**:

`git grep isNodeConditionSetAsExpected` :)

### PR DESCRIPTION
**Pull request by mattfenwick**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

The IsNodeConditionSetAsExpected function is duplicated in two places, this PR moves the implementation into testutils where both callsites can use the common implementation.

**Which issue(s) this PR fixes**:

Fixes https://github.com/kubernetes/kubernetes/issues/93965

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

### CODE REVIEW

File: test/utils/node.go [hunk 1]
```diff
@@ -31,3 +51,63 @@ func GetNodeCondition(status *v1.NodeStatus, conditionType v1.NodeConditionType)
...
+					msg := ""
+					if !hasNodeControllerTaints {
+						msg = fmt.Sprintf("Condition %s of node %s is %v instead of %t. Reason: %v, message: %v",
+							conditionType, node.Name, cond.Status == v1.ConditionTrue, wantTrue, cond.Reason, cond.Message)
+					} else {
+						msg = fmt.Sprintf("Condition %s of node %s is %v, but Node is tainted by NodeController with %v. Failure",
+							conditionType, node.Name, cond.Status == v1.ConditionTrue, taints)
+					}
+					if !silent {
+						klog.Infof(msg)
```
Comment (mattfenwick) [inline test/utils/node.go:84]: The only difference between the two implementations is calls to `e2elog` vs calls to `klog` , I saw other places in the codebase where the logging func was parameterized so I'm going to try to do the same thing here.

**Comment by jayunit100** [comment]
yup this is what i think needs to be done, will wait for ixdy to confirm

**Comment by dims** [comment]
@mattfenwick there are efforts to separate out test/e2e/framework as a standalone / staging repo so this would be counter to the effort i think

**Comment by mattfenwick** [comment]
@dims do you want me to close this PR or try to merge this into the test/e2e/framework separation effort?  Fine with either option!

**Comment by dims** [comment]
@mattfenwick yes please close this (let's make sure test/e2e/framework can be standalone)

**Comment by mattfenwick** [comment]
Closing based on @dims [comment](https://github.com/kubernetes/kubernetes/pull/93983#issuecomment-674528771).

**Pull request closed without merging**